src/types/typeGuards/objects.ts

```diff
@@ -0,0 +1,67 @@
+import * as ts from "typescript";
+
+import { isObjectFlagSet } from "../../flags.js";
+import { isGenericType } from "./compound.js";
+import { isObjectType } from "./simple.js";
+
+export function isEvolvingArrayType(
+	type: ts.Type
+): type is ts.EvolvingArrayType {
+	return (
+		isObjectType(type) && isObjectFlagSet(type, ts.ObjectFlags.EvolvingArray)
+	);
+}
+
+export function isInterfaceType(type: ts.Type): type is ts.InterfaceType {
```

Comment:
This type is really confusing me. We need to look into this more.

The comment in the TypeScript type declarations file seems to imply that `ObjectFlags.Class` or `ObjectFlags.Interface` should be set on any instance of this type.

However, if this type supposed to include types declared by an interface, then the fields on `InterfaceType` don't line up with the runtime data; so `InterfaceType` must not include such types (hence [this test](https://github.com/RebeccaStevens/ts-api-utils/blob/db5958611b42655486e47b4ef01c34f32695f747/src/types/typeGuards/objects.test.ts#LL22C3-L22C45).
Classes and instances of classes do have the right runtime data though. They also have the `ObjectFlags.Class` object flag; so everything is right for them.

But there are types that they are an `InterfaceType` but though have either `ObjectFlags.Class` or `ObjectFlags.Interface` set. For example, `TupleType` extends `GenericType` which extends `InterfaceType`.

Reply:
I've remove this function for now. We can look into it again if future.